lib/time.py

```diff
@@ -23,3 +55,39 @@ def sleep(secs):
 
 def time():
   return float(Now().UnixNano()) / Second
+
+
+_strftime_directive_map = {
+    'Y': '2006', 'y': '06', 'b': 'Jan', 'B': 'January', 'm': '01', 'd': '02',
+    'a': 'Mon', 'A': 'Monday',
+    'H': '15', 'I': '03', 'M': '04', 'S': '05', 'L': '.000',
+    'p': 'PM',
+    'Z':  'MST', 'z': '-0700',
+    '%': '%',
+}
+
+def strftime(format, tt=None): #pylint: disable=missing-docstring
+  t = (Unix(int(mktime(tt)), 0) if tt else Now()).Local()
+  ret = []
+  prev, n = 0, format.find('%', 0, -1)
+  while n != -1:
+    ret.append(format[prev:n])
+    next_ch = format[n+1]
+    if next_ch in 'cjUwWxX0123456789':
```

Comment:
Argh. Thanks for clarifying. It's annoying how inconsistent it all is:

- According to [man strftime](http://man7.org/linux/man-pages/man3/strftime.3.html), Linux supports padding and 0 flag for zero-padding
- [The UNIX spec](http://pubs.opengroup.org/onlinepubs/007908799/xsh/strftime.html) does not mention any kind of padding
- [GNU strftime](http://search.cpan.org/~dexter/POSIX-strftime-GNU-0.02/lib/POSIX/strftime/GNU.pm) supports 0 padding but not arbitrary length padding (same as for [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=strftime&sektion=3) which explains the way OS X behaves)
- And of course [time.strftime()](https://docs.python.org/2/library/time.html#time.strftime) defers to the platform specific behavior

Thinking about this some more, I'm inclined to make this as simple as possible while still adhering to the Python spec. I don't see any value in supporting any extensions specific to one particular platform.

As far as whether '%f' -> 'f' or '%f' -> '%f', I don't have a strong opinion but I'd probably lean toward Linux since it's so widely used for server development.